src/Patchwork/PHP/Shim/Mbstring.php

```diff
@@ -271,7 +279,10 @@ public static function mb_detect_encoding($str, $encoding_list = INF, $strict =
                     break;
 
                 default:
-                    return strncmp($enc, 'ISO-8859-', 9) ? false : $enc;
+                    if (0 === strncmp($enc, 'ISO-8859-', 9)) {
+                        return $enc;
+                    }
+                    break;
```

Comment:
```php
array('ASCII', 'ISO-2022-JP', 'UTF-8', 'ISO-8859-1')
```

Without the change, the method would return `false` for `ISO-2022-JP`, even though the list contains more, valid charsets. After the change, the unknown charset is ignored, so the list can be run trough until the end and return the correct charset.